utils/regiondetection.py

```diff
@@ -1,243 +1,314 @@
-        return False
-    if (x_percent > 87.891): #x_percent < 5.859 or 
-        return False
-    if (y_percent > 95.052): #y_percent < 2.604 or
-        return False
-
-    # Example of the colour count and dominant colour percentage
-    ccnt, pct = count_colours(region)
-    if pct < 4 or pct > 82:
-        return False
-    if ccnt < 30 or ccnt > 8572:
-        return False
-
-    # OCR can be added to the filter like this:
-    regionT = region[margin:-margin, margin:-margin]
-    text = getText(regionT).strip()
-    if len(text) > 2:
-        if checkBlacklist(text, 3):
-            return False
-
-    text = getText(regionT,False).strip()
-    if len(text) > 2:
-        if checkBlacklist(text, 3):
-            return False
-    return True
-
-def _findregions(image, imgpath, draw=True, highlightname="Highlight", invert=True):
-    "Finds ALL regions in the linked image"
-
-    if draw:
-        drawimg = np.copy(image)
-
-    main_logger.debug("Obtaining grayscale version of image")
-    img = cv2.cvtColor(image,cv2.COLOR_BGR2GRAY)
-    if draw:
-        cv2.imwrite(f"{highlightname}-0-grey.png", img)
+# Draw the detected regions on the originial image
+def _draw_regions(image: cv2.typing.MatLike, img_path: str, regions: list, highlight_name: str):
+        draw_img = np.copy(image)
         
+        for index, region in enumerate(regions):
+            main_logger.debug(f"Drawing region #{index}")
+            
+            region_height, region_width, _ = region[0].shape
+            x = region[2]
+            y = region[3]
+            
+            color_int = random.randint(0, 2)
+            colors = [(0,0,255), (0,255,0), (255,0,0)]
+            color = colors[color_int]
+            
+            flip = (random.randint(0, 1) == 1)
+            
+            cv2.rectangle(draw_img, (x - 5, y - 5), (x + region_width - 5, y + region_height - 5), color, 1)
+            
+            if region[7]:
+                text = "-" + str(region[1])
+            else:
+                text = str(region[1])
+                
+            if flip:
+                cv2.putText(draw_img, text, (x + region_width - random.randint(-5, 5), y + region_height-random.randint(-5, 5)), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2)
+            else:
+                cv2.putText(draw_img, text, (x - random.randint(-5, 5), y - random.randint(-5, 5)), cv2.FONT_HERSHEY_SIMPLEX, 0.5, color, 2)
+    
+            # TODO: Add subregion drawing
+            #if subregiondraw:
+            #     cv2.imwrite(f"{highlightname}.subregion.{idx}.png", region[0])
+            
+        cv2.imwrite(os.path.join(os.path.dirname(os.path.realpath(img_path)), f"{highlight_name}.png"), draw_img)
+
+# TODO: Add drawing capabilities for recursive and try to split the function into smaller parts
+def _find_regions(image : cv2.typing.MatLike, image_path : str, draw : bool, highlight_name : str, invert = True):
+    
+    main_logger.debug("Obtaining grayscale version of image")

```

Comment:
I think it'd be nice to seperate this image pre-processing from the region detection post-processing

Reply:
Agree